src/wallet/connection.ts

```diff
@@ -88,4 +88,32 @@ class WalletServiceConnection extends BaseConnection {
   }
 }
 
-export default WalletServiceConnection;
+/**
+ * This connection is used when the wallet has websocket disabled.
+ */
+export class DummyWalletServiceConnection extends BaseConnection {
```

Comment:
I'm thinking about this dummy class. I like it in general, to be honest, but it's missing many method. For instance, the wallet class calls `this.conn.stop()` and this dummy class is not overwriting the `BaseConnection` `stop` method, so we would probably get an error.

We also have many public methods in the `BaseConnection`, e.g., `onConnectionChange`, `setState`, `endConnection`, `setup`, ...

I understand that those method shouldn't be used but they are public and we might be using them directly in the wallets.

Not sure what's the best approach here, to be honest, you could overwrite all those method but, if we add a new one to the `BaseConnection` in the future, we would need to remember to also overwrite here, that's why I'm not sure if this is the best thing to do.

What do you think about it?

Reply:
Humm... we can extract an interface for the BaseConnection, this way we do not forget, and for the methods not being used, we can throw errors or do nothing.

This is a class for a very specific case, we can stress it in the class documentation and make it clear it should not be used for other purposes.

The other way to solve it is by doing what is specified in the design, but it interferes more with the wallet code and I avoided it on purpose.